cms/auth/checks.py

```diff
@@ -0,0 +1,95 @@
+            ("AWS_REGION", "eu-west-2", "auth.E003"),
+            ("AUTH_TOKEN_REFRESH_URL", "http://localhost:29500/tokens/self", "auth.E004"),
+            ("LOGOUT_REDIRECT_URL", "http://localhost:29500/logout", "auth.E005"),
+        ]
+        for setting, hint_value, error_id in cognito_settings:
+            value = getattr(settings_obj, setting, None)
+            if not value:
+                errors.append(
+                    Error(
+                        f"{setting} is required when AWS_COGNITO_LOGIN_ENABLED is True.",
+                        hint=f"Set {setting} to an appropriate value. e.g. '{hint_value}'.",
+                        id=error_id,
+                    )
+                )
+
+
+def _check_session_config(errors: list, settings_obj: "LazySettings") -> None:
+    session_cookie_age = getattr(settings_obj, "SESSION_COOKIE_AGE", None)
+    session_renewal_offset = getattr(settings_obj, "SESSION_RENEWAL_OFFSET_SECONDS", None)
+    if session_cookie_age and session_renewal_offset and session_renewal_offset >= session_cookie_age:
+        errors.append(
+            Error(
+                f"SESSION_RENEWAL_OFFSET_SECONDS ({session_renewal_offset}) should be less than "
+                f"SESSION_COOKIE_AGE ({session_cookie_age}).",
+                hint="Set SESSION_RENEWAL_OFFSET_SECONDS to a value smaller than SESSION_COOKIE_AGE.",
+                id="auth.E006",
+            )
+        )
+
+
+def _check_identity_api(errors: list, settings_obj: "LazySettings") -> None:
+    identity_api_url = getattr(settings_obj, "IDENTITY_API_BASE_URL", None)
+    team_sync_enabled = getattr(settings_obj, "AWS_COGNITO_TEAM_SYNC_ENABLED", False)
+    if identity_api_url and team_sync_enabled:
+        service_auth_token = getattr(settings_obj, "SERVICE_AUTH_TOKEN", None)
+        if not service_auth_token:
+            errors.append(
+                Error(
+                    (
+                        "SERVICE_AUTH_TOKEN is not set but IDENTITY_API_BASE_URL is configured and "
+                        "AWS_COGNITO_TEAM_SYNC_ENABLED is True."
+                    ),
+                    hint="Set SERVICE_AUTH_TOKEN for API authentication.",
+                    id="auth.E007",
+                )
+            )
+
+
+def _check_team_sync(errors: list, settings_obj: "LazySettings") -> None:
+    team_sync_enabled = getattr(settings_obj, "AWS_COGNITO_TEAM_SYNC_ENABLED", False)
+    if team_sync_enabled:
+        team_sync_frequency = getattr(settings_obj, "AWS_COGNITO_TEAM_SYNC_FREQUENCY", None)
+        if team_sync_frequency is not None and team_sync_frequency < 1:
+            errors.append(
+                Error(
+                    f"AWS_COGNITO_TEAM_SYNC_FREQUENCY must be at least 1 minute (got {team_sync_frequency}).",
+                    hint="Set AWS_COGNITO_TEAM_SYNC_FREQUENCY to a positive integer.",
+                    id="auth.E008",
+                )
+            )
```

Comment:
Another candidate for flattening:
```suggestion
    if not getattr(settings_obj, "AWS_COGNITO_TEAM_SYNC_ENABLED", False):
        return None
    team_sync_frequency = getattr(settings_obj, "AWS_COGNITO_TEAM_SYNC_FREQUENCY", None)
    if team_sync_frequency is not None and team_sync_frequency < 1:
        errors.append(
            Error(
                f"AWS_COGNITO_TEAM_SYNC_FREQUENCY must be at least 1 minute (got {team_sync_frequency}).",
                hint="Set AWS_COGNITO_TEAM_SYNC_FREQUENCY to a positive integer.",
                id="auth.E008",
            )
        )
```

Reply:
Done! [4fc9f38](https://github.com/ONSdigital/dis-wagtail/pull/300/commits/4fc9f381bf9f26a40d8d225f178ef536d190517c)